nixos/modules/services/backup/restic.nix

```diff
@@ -31,6 +31,59 @@ in
           '';
         };
 
+        rcloneOptions = mkOption {
+          type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+          default = null;
+          description = ''
+            Options to pass to rclone to control its behavior.
+            See <link xlink:href="https://rclone.org/docs/#options"/> for
+            available options. When specifying option names, strip the
+            leading <literal>--</literal>. To set a flag such as
+            <literal>--drive-use-trash</literal>, which does not take a value,
+            set the value to <literal>"true"</literal>.
+          '';
+          example = {
+            bwlimit = "10M";
+            drive-use-trash = "true";
+          };
+        };
+
+        rcloneConfig = mkOption {
+          type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+          default = null;
+          description = ''
+            Configuration for the rclone remote being used for backup.
+            See the remote's specific options under rclone's docs at
+            <link xlink:href="https://rclone.org/docs/"/>. When specifying
+            option names, use the "config" name specified in the docs. 
+            For example, to set <literal>--b2-hard-delete</literal> for a B2
+            remote, use <literal>hard_delete = "true"</literal> in the
+            attribute set.
+            Warning: Secrets set in here will be world-readable in the Nix
+            store! Consider using the <literal>rcloneConfigFile</literal>
+            option instead to specify secret values, which will take precedence
+            over options set here.
```

Comment:
How well does using a config file and using these options play together?

Reply:
The environment variables essentially set the "default" value for rclone, but if it finds that value also set in the config file, rclone will use the value in the latter.

For example, if I specified `RCLONE_CONFIG_REMOTEA_TYPE = "s3"`, but then in the config file I had
```
[remoteA]
type=b2
```
rclone will treat `remoteA` as a B2 remote.

Reply:
Is this intended behaviour? I'd doubt. Maybe open a bug upstream and don't specify it here (or at least link to the bug).

We shouldn't mark this as a feature, so people don't rely on it.

Reply:
I opened a question on their forums [here](https://forum.rclone.org/t/precedence-rules-for-config/17707) to confirm this is intended behavior. If it turns out I'm not supposed to do this (or that we shouldn't rely on it regardless), I can either

1. Remove `rcloneConfig` and just have users use `rcloneConfigFile` on its own, or
2. Keep both options, but add an assertion to the module that only one option can be used at a time. Users can use one option or the other, but not both.

Which would be preferable?

Reply:
https://rclone.org/docs/#environment-variables reads like environment variables should set the default, but not win over command line arguments or config files.

Given rclone also supports [config file encryption](https://rclone.org/docs/#configuration-encryption), we should definitely keep `rcloneConfigFile`.

I'm not sure if we need both `rcloneConfig` and `rcloneOptions` (if they provide the same feature set). WDYT?

Reply:
Yeah, that wording in their docs wasn't the best. Hopefully they'll clear things up. I agree we should keep the `rcloneConfigFile` option.

`rcloneConfig` and `rcloneOptions` provide different things. `rcloneOptions` was meant to control global rclone behavior, i.e. options that are listed at https://rclone.org/docs/#options. This is stuff like bandwidth limits, using checksums to compare files, etc. `rcloneConfig` is used to configure specific remotes, e.g. S3 access credentials, etc, i.e. stuff available in the docs for specific remote types (for example, https://rclone.org/s3/#standard-options for S3 remotes).

Reply:
[Looks like this is intended behavior.](https://forum.rclone.org/t/precedence-rules-for-config/17707/2?u=i077) See linked post for the order of precedence.

Reply:
@flokli if you're good with keeping both options, I'm going to resolve this thread.